include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -530,6 +530,17 @@ def quake_ApplyNoiseOp : QuakeOp<"apply_noise", [AttrSizedOperandSegments]> {
   }];
 }
 
+def quake_SaveStateOp : QuakeOp<"save_state"> {
```

Comment:
I don't think there are any optimizations planned for this and therefore it shouldn't be added to the IR. It should suffice to lower this directly to a CallOp to a runtime library function that is a common decl across all simulators.